server/py/services/api/initial_data.py

```diff
@@ -993,6 +1002,46 @@ def _create_project_summaries(db, db_session):
     db._upsert(db_session, project_summaries, ignore=True)
 
 
+def _init_system_id(db_session: sqlalchemy.orm.Session):
```

Comment:
What's missing from this function is setting the system id in the config.
Currently you only ensure it is in the DB.
You want it to be available in `mlconf.system_id`, so you need to make sure that at you populate it with the resolved value (overwrite it if needed).